server/controller/message/communityMessage.js

```diff
@@ -49,6 +49,55 @@ class CommunityMessageController {
       next(error);
     }
   }
+
+  /**
+   *
+   * @param {object} req request object
+   * @param {object} res response object
+   * @param {object} next calls next object
+   * @returns {object} returns
+   */
+  static async deleteMessage(req, res, next) {
+    try {
+      const { userId } = req.user;
+      const user = await User.findByPk(userId);
+      const { id } = req.params;
+      const message = await CommunityMessage.findAll({
+        attributes: ['senderId'],
+        where: {
+          id,
+        }
+      });
+      if (user) {
```

Comment:
let's have this check first before we even begin fetching the message, like this

```js
// start by checking if we have a user first
if (! user) {
  ...
}

// then we fetch the message
const message = await CommunityMessage.findAll({
  ...
})

// then we proceed to check whether there's a message
if (! message.length) {
  ...
}
```
By doing it this way, we prevent querying the DB for `communityMessage` when there's no `user` hence, saving us some runtime. Because when that condition (`if (! user) {...}`) passes it returns and stops the rest of the code from executing.

Reply:
@Ochowo let me know if this makes sense ☝️

Reply:
okay@fortune will look into it now